tests/helpers/FakeEthLightwallet.ts

```diff
@@ -0,0 +1,294 @@
+   */
+  public setError (methodName) {
+    this.errors[methodName] = true
+  }
+
+  /**
+   * Removes flags to mock errors for all methods.
+   */
+  public removeErrors () {
+    this.errors = []
+  }
+
+  /**
+   * Mocked lightwallet.txutils methods
+   */
+  private _functionTx (
+    abi,
+    functionName,
+    args,
+    options
+  ) {
+    if (this.errors['functionTx']) {
+      throw new Error('Mocked error in lightwallet.txutils.functionTx()')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+  private _valueTx (options) {
+    if (this.errors['valueTx']) {
+      throw new Error('Mocked error in lightwallet.txutils.valueTx()')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+
+  /**
+   * Mocked lightwallet.signing methods
+   */
+  private _signTx (
+    keystore,
+    pwDerivedKey,
+    rlpHexTx,
+    address
+  ) {
+    if (this.errors['signTx']) {
+      throw new Error('Mocked error in lightwallet.signing.signTx')
+    }
+    return '0xf86180808401ef364594f0109fc8df283027b6285cc889f5aa624eac1f5580801ca031573280d608f75137e33fc14655f097867d691d5c4c44ebe5ae186070ac3d5ea0524410802cdc025034daefcdfa08e7d2ee3f0b9d9ae184b2001fe0aff07603d9'
+  }
+  private _signMsgHash (
+    keystore,
+    pwDerivedKey,
+    personalMsgHashBuff,
+    address
+  ) {
+    if (this.errors['signMsgHash']) {
+      throw new Error('Mocked error in lightwallet.signing.signMsgHash')
+    }
+    return {
+      r: new Buffer('r'),
+      s: new Buffer('s'),
+      v: 24
+    }
+  }
+  private _concatSig (signature) {
+    return 'Concat signature'
+  }
+
+  /**
+   * Mocked lightwallet.keystore methods
+   */
+  private _serialize () {
+    return keystore1
+  }
+  private _deserialize (serializedKeystore) {
+    return this.keystore
+  }
+  private _keyFromPassword (password, callback) {
+    if (this.errors['keyFromPassword']) {
+      return callback(new Error('Mocked error in lightwallet.keystore.keyFromPassword()'), undefined)
+    }
+    return callback(undefined, this._pwDerivedKey)
+  }
+  private _exportPrivateKey (address, pwDerivedKey) {
+    if (this.errors['exportPrivateKey']) {
+      throw new Error('Mocked error in lightwallet.keystore.exportPrivateKey')
+    }
+    return this._privateKey
+  }
+  private _createVault (options, callback) {
+    if (this.errors['createVault']) {
+      return callback(new Error('Mocked error in lightwallet.keystore.createVault()'), undefined)
+    }
+    return callback(undefined, this.keystore)
+  }
+  private _generateRandomSeed () {
+    if (this.errors['generateRandomSeed']) {
+      throw new Error('Mocked error in lightwallet.keystore.generateRandomSeed()')
```

Comment:
I wanted to make sure that every possible error thrown by eth-lightwallet is properly caught. I looked into the source code of `generateRandomSeed()` and it can only throw an error when it is called with an optional argument `extraEntropy`. But as we don't use the function with an argument, it doesn't really make sense to expect an error here... Will remove it

Reply:
Okay, then you would need to try and catch it and then call reject with the error.